chromadb/segment/impl/manager/local.py

```diff
@@ -112,6 +112,9 @@ def delete_segments(self, collection_id: UUID) -> Sequence[UUID]:
         segments = self._sysdb.get_segments(collection=collection_id)
         for segment in segments:
             if segment["id"] in self._instances:
+                if segment["type"] == SegmentType.HNSW_LOCAL_PERSISTED.value:
+                    instance = self.get_segment(collection_id, VectorReader)
+                    instance.reset_state()
```

Comment:
I don't think we should be using reset_state() here. inside of delete_segments. allow_reset=False could theoretically allow that to error if the implementation of a segment wanted to. Also, the base class intent of this method is to

```
Delete any local state for all the segments associated with a collection, and
returns a sequence of their IDs. Does not update the SysDB.
```
        
I think we are reaching out past what this method should touch. I think perhaps a segment should have a delete()

Reply:
Agreed; this makes sense. Let me update the interface to allow for delete() method.

Reply:
@HammadB, do you think it's OK to define the abstract delete op at `LocalHnswSegment` level so that it can only apply to the local segment? I thought of adding it all the way up in `Segment`, but that has a fair bit of downstream implementations which may not need this.

Reply:
nvm, I think it would work in `SegmentImplementation`. I've just added the methods to `SqliteMetadataSegment` and `LocalHnswSegment` as they are generalizations and made them throw NotImplemented. In the future if we find use of delete in these specific segments we can always add implementation.

As far as the description goes I think it should be fine as is. We do indeed delete an and all local state associated with a collection which in fact is meta + hnsw, correct?

Last but not least:

```python
 @override
    def reset_state(self) -> None:
        data_path = self._get_storage_folder()
        if os.path.exists(data_path):
            shutil.rmtree(data_path, ignore_errors=True)

````

I think we should add a check for `settings.allow_reset`.

Reply:
1. Yes we should add a check for the allow_reset
2. For the description. The delineation is that the comment is "local" state but the segment may be remote. For example imagine a segment implementation that is a grpc client to a remote segment (this is how it will work in the distributed case).